### PR TITLE
Rearrange modules

### PR DESCRIPTION
**Pull request by mr-archano**
This quite noisy PR focuses on two main things:
- Restructure the folders inside the repo to have the plugin source code inside a `plugin/` folder (actually `plugin/core` because I am planning to add more modules, more on that soon)
- Use composite builds in order to include the plugin project in `samples`. This will ensure that the `samples` will always use a locally compiled version of the plugin, so it will make it easier to experiment with the plugin sourcecode.

### CODE REVIEW

**Comment by mr-archano** [comment]
@hal90002 test this please

**Comment by mr-archano** [comment]
@hal90002 retest this please

File: samples/settings.gradle [hunk 1]
```diff
@@ -1,2 +1,9 @@
+rootProject.name = 'bintray-release-samples'
+
 include ':jvm', ':android'
-rootProject.name = 'bintray-release-samples'
\ No newline at end of file
+
+includeBuild('../') {
+    dependencySubstitution {
+        substitute module('com.novoda:bintray-release:local') with project(':core')
```
Comment (ouchadam) [inline samples/settings.gradle]: :100:

**Comment by blundell** [comment]
eyy +1 on not having to release a version to use in the sample, always complicated things :)

**Pull request merged**